modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/internal/TcpDiscoveryNodesRing.java

```diff
@@ -260,12 +279,8 @@ public long maxInternalOrder() {
         rwLock.readLock().lock();
 
         try {
-            if (maxInternalOrder == 0) {
```

Comment:
Right now nodes.last() don't given nodes with maximum of internal order, because I changed order. So I update maxInternalOrder more often in other place. Also nodes.last()==null branch is unreachable.